File: wardy/transfer_token.mjs
import { Api, JsonRpc, RpcError, JsSignatureProvider } from '../dist/index.js'
import * as dotenv from 'dotenv' 

dotenv.config()

const url = process.env.NODE_URL; // use your http_address with port : 88888
const json_rpc = new JsonRpc(url); // create new JsonRPC using your node url


const private_key = process.env.PRIVATE_KEY; // private key of your inery account
const account = process.env.INERY_ACC; //your account inery
const actor = process.env.INERY_ACC; // The Signer
const token = process.env.TOKEN_TRANSFER;
const signature  = new JsSignatureProvider([private_key]); // create signature from private_key
const api = new Api({
    rpc: json_rpc,
    signatureProvider: signature
});

// tranfer Function on Smart Contract
async function transfertoken(from, to, quantity, memo){
    try{
        const tx = await api.transact({
            actions:[
                {
                  account,
                  name:"transfer",
                  authorization:[
                        {
                            actor,
                            permission:"active"
                        }
                    ],
                    data:{
                        from, to, quantity, memo
                    }
                }
            ]
        },{broadcast:true,sign:true})

        console.log(tx) // output the tx to terminal, it's Json Object
        console.log(tx.processed.action_traces[0].console)
    }catch(error){
        console.log(error)
    }
}

transfertoken(actor, "inery", token, "Free From RPC")

File: arsayya/transfer_token.py
import os
from api.cline import Cline
from api.keys import INRKey
from ineryjs import Api, JsonRpc, RpcError, JsSignatureProvider

os.environ.update({k.lower(): v for k, v in os.environ.items()})

TOKEN_CONTRACT = os.getenv("TOKEN_CONTRACT")
FROM_ACCOUNT = os.getenv("SENDER_ACCOUNT")
TO_ACCOUNT = os.getenv("RECIPIENT_ACCOUNT")
TOKEN_SYMBOL = os.getenv("TOKEN_SYMBOL")
AMOUNT = os.getenv("TRANSFER_AMOUNT")
MEMO = os.getenv("TRANSFER_MEMO")

json_rpc = JsonRpc(os.getenv("INERY_NODE_RPC"))
signature_provider = JsSignatureProvider([os.getenv("PRIVATE_KEY")])

api = Api({
    "rpc": json_rpc,
    "signature_provider": signature_provider
})

async def transfer_token(from_account, to_account, amount, memo):
    try:
        tx = await api.transact({
            "actions": [
                {
                    "account": TOKEN_CONTRACT,
                    "name": "transfer",
                    "authorization": [
                        {
                            "actor": from_account,
                            "permission": "active"
                        }
                    ],
                    "data": {
                        "from": from_account,
                        "to": to_account,
                        "quantity": amount,
                        "memo": memo
                    }
                }
            ]
        }, {"broadcast": True, "sign": True})

        print(tx["processed"])
        print(tx["processed"]["action_traces"][0]["act"])
    except RpcError as error:
        print(f"ERROR: Can't transfer token symbol {TOKEN_SYMBOL}")
        print(f"DETAILS: {error.details[0]['message']}")

transfer_token(FROM_ACCOUNT, TO_ACCOUNT, AMOUNT, MEMO)
